From 7edb4140804c1a9054f952c383ae6b34b4fa96e0 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 24 May 2026 21:06:23 +0800
Subject: [PATCH 332/344] AOSCOS: rust: Add -fpatchable-function-entry= to
 bindgen_skip_c_flags

This flag was introduced with commit eec37961a56a ("powerpc64/ftrace: Move
ftrace sequence out of line") to workaround an issue with ftrace. However,
this flag was passed to bindgen and caused the following error:

  Unable to generate bindings: clang diagnosed error: error: unsupported option '-fpatchable-function-entry=1' for target 'powerpc64le-unknown-linux-gnu'

Add this flag to bindgen_skip_c_flags to fix build.

Fixes: "BACKPORT: FROMLIST: powerpc: Enable Rust for ppc64le"
Tested-by: Mingcong Bai <jeffbai@aosc.io>
---
 rust/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rust/Makefile b/rust/Makefile
index 22dac1cfec412..e75c7d159ea2e 100644
--- a/rust/Makefile
+++ b/rust/Makefile
@@ -445,7 +445,7 @@ bindgen_skip_c_flags := -mno-fp-ret-in-387 -mpreferred-stack-boundary=% \
 	-fzero-init-padding-bits=% -mno-fdpic \
 	-fdiagnostics-show-context -fdiagnostics-show-context=% \
 	--param=% --param asan-% -fno-isolate-erroneous-paths-dereference \
-	-mno-riscv-attribute
+	-mno-riscv-attribute -fpatchable-function-entry=%
 
 # All warnings are inhibited since GCC builds are very experimental,
 # many GCC warnings are not supported by Clang, they may only appear in
-- 
2.52.0

